### PR TITLE
making NumSnapshotsToKeep configurable

### PR DESCRIPTION
**Pull request by sreekanth-cb**
NumSnapshotsToKeep read from index configs

### CODE REVIEW

File: index/scorch/persister.go [hunk 1]
```diff
@@ -633,14 +633,19 @@ func (s *Scorch) removeOldBoltSnapshots() (numRemoved int, err error) {
 		return 0, err
 	}
 
-	if len(persistedEpochs) <= NumSnapshotsToKeep {
+	numSnapshotsToKeep := NumSnapshotsToKeep
+	if val, ok := s.config["numSnapshotsToKeep"].(float64); ok && val > 0 {
```
Comment (mschoch) [inline index/scorch/persister.go]: i think this keeps coming up from time to time.  from the Go library perspective this is kinda ugly as users who pass an integer (what you would expect) find that the feature doesn't work.  i think if we add these kind of options, they should make sense from the Go library perspective first, even if that means more work elsewhere to parse JSON differently...

File: index/scorch/scorch.go [hunk 2]
```diff
@@ -191,6 +192,23 @@ func (s *Scorch) openBolt() error {
 		}
 	}
 
+	s.numSnapshotsToKeep = NumSnapshotsToKeep
```
Comment (mschoch) [inline index/scorch/scorch.go]: hmmn, i don't think round tripping the value through json is the right approach either

File: index/scorch/scorch.go [hunk 3]
```diff
@@ -503,3 +515,15 @@ func (s *Scorch) unmarkIneligibleForRemoval(filename string) {
 func init() {
 	registry.RegisterIndexType(Name, NewScorch)
 }
+
+func parseToInteger(v interface{}) (int, error) {
+	switch v.(type) {
+	case float64:
+		return int(v.(float64)), nil
```
Comment (mschoch) [inline index/scorch/scorch.go]: Go has syntax to allow remembering the newly typed value in a type assertion, so you don't have to do the assertion a second time when returning the value here.  See this example: https://tour.golang.org/methods/16

Comment (sreekanth-cb) [reply index/scorch/scorch.go]: done!

**Pull request merged**